packages/plugins/plugin-sheet/src/types.ts

```diff
@@ -21,13 +22,24 @@ export enum SheetAction {
   CREATE = `${SHEET_ACTION}/create`,
 }
 
+// TODO(Zan): Move this to the plugin-space plugin or another common location
+// when we implement comments in sheets.
+// This is currently duplicated in a few places.
+type ThreadProvides<T> = {
+  thread: {
+    predicate: (obj: any) => obj is T;
```

Comment:
document (also see comment re MarkdownPlugin that deviates from this signature).